File: LogViewer/src/com/logviewer/model/MatchResult.java
package com.logviewer.model;

/**
 * Pass 1 에서 발견된 매칭 라인 하나를 나타내는 객체.
 * lineNumber : 원본 파일 기준 1-based 줄 번호
 * matchedKeyword : 해당 라인을 매칭시킨 키워드 (날짜 조건 단독 매칭 시에는 날짜 문자열)
 */
public class MatchResult {

    private final int    lineNumber;
    private final String matchedKeyword;

    public MatchResult(int lineNumber, String matchedKeyword) {
        this.lineNumber     = lineNumber;
        this.matchedKeyword = matchedKeyword;
    }

    // Python: int
    // TS: number
    public int    getLineNumber()       { return lineNumber; }
    // Python: str
    // TS: string
    public String getMatchedKeyword()   { return matchedKeyword; }
}

File: LogViewer/src_py/model/match_result.py
"""
Java 대응: MatchResult (LogExtractService의 Inner Class)

변환 포인트:
    final int    lineNumber     →  line_number:     int   (불변 필드)
    final String matchedKeyword →  matched_keyword: str   (불변 필드)

    Java에서 final = 재할당 불가
    Python에서는 @dataclass(frozen=True) 로 동일 효과
"""
from dataclasses import dataclass


@dataclass(frozen=True)   # frozen=True → Java의 final 필드와 동일 (변경 불가)
class MatchResult:
    """
    Pass 1 에서 발견된 매칭 라인 하나를 나타내는 객체.

    Java 생성자 대응:
        new MatchResult(lineNumber, matchedKeyword)
        →  MatchResult(line_number=5, matched_keyword="ERROR")
           또는 MatchResult(5, "ERROR")
    """
    line_number:     int   # Java: final int    lineNumber
    matched_keyword: str   # Java: final String matchedKeyword
